benchmark/CMakeLists.txt

```diff
@@ -4,6 +4,11 @@ if (NOT CMAKE_BUILD_TYPE STREQUAL "Release")
         "will be affected")
 endif()
 
+if (GINKGO_BUILD_CUDA AND GINKGO_BUILD_HIP AND GINKGO_HIP_PLATFORM MATCHES "hcc")
+  message(FATAL_ERROR "Building the benchmarks with both HIP AMD and CUDA support "
+      "at the same time is currently not supported. Use `export HIP_PLATFORM=nvcc` instead.")
```

Comment:
nit: `With both HIP AMD and CUDA support [...] is not supported` sounds a bit weird.
maybe:
`Building the benchmarks for both HIP AMD and CUDA at the same time is currently not supported.`